File: Quiz PBO/appkereta (sprinthost.ru SERVER)/Tiket.php
<?php
require_once 'database.php';
class Tiket 
{
    private $db;
    private $table = 'tiket';
    public $id_pemesan = "";
    public $id_kereta = "";
    public $no_kursi = "";
    public $jadwal = "";
    public $kelas = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_id_pemesan(int $id_pemesan)
    {
        $query = "SELECT * FROM $this->table WHERE id_pemesan = $id_pemesan";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`id_pemesan`,`id_kereta`,`no_kursi`,`jadwal`,`kelas`) VALUES ('$this->id_pemesan','$this->id_kereta','$this->no_kursi','$this->jadwal','$this->kelas')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET id_pemesan = '$this->id_pemesan', id_kereta = '$this->id_kereta', no_kursi = '$this->no_kursi', jadwal = '$this->jadwal', kelas = '$this->kelas' 
        WHERE id_tiket = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_id_pemesan($id_pemesan): int
    {
        $query = "UPDATE $this->table SET id_pemesan = '$this->id_pemesan', id_kereta = '$this->id_kereta', no_kursi = '$this->no_kursi', jadwal = '$this->jadwal', kelas = '$this->kelas' 
        WHERE id_pemesan = $id_pemesan";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_tiket = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_id_pemesan($id_pemesan): int
    {
        $query = "DELETE FROM $this->table WHERE id_pemesan = $id_pemesan";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Quiz PBO/GUI Form Kereta/Tiket.py
import requests
import json
class Tiket:
    def __init__(self):
        self.__id=None
        self.__id_pemesan = None
        self.__id_kereta = None
        self.__no_kursi = None
        self.__jadwal = None
        self.__kelas = None
        self.__url = "http://localhost/appkereta/tiket_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def id_pemesan(self):
        return self.__id_pemesan
        
    @id_pemesan.setter
    def id_pemesan(self, value):
        self.__id_pemesan = value
    @property
    def id_kereta(self):
        return self.__id_kereta
        
    @id_kereta.setter
    def id_kereta(self, value):
        self.__id_kereta = value
    @property
    def no_kursi(self):
        return self.__no_kursi
        
    @no_kursi.setter
    def no_kursi(self, value):
        self.__no_kursi = value
    @property
    def jadwal(self):
        return self.__jadwal
        
    @jadwal.setter
    def jadwal(self, value):
        self.__jadwal = value
    @property
    def kelas(self):
        return self.__kelas
        
    @kelas.setter
    def kelas(self, value):
        self.__kelas = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_id_pemesan(self, id_pemesan):
        url = self.__url+"?id_pemesan="+id_pemesan
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_tiket']
            self.__id_pemesan = item['id_pemesan']
            self.__id_kereta = item['id_kereta']
            self.__no_kursi = item['no_kursi']
            self.__jadwal = item['jadwal']
            self.__kelas = item['kelas']
        return data
    def simpan(self):
        payload = {
            "id_pemesan":self.__id_pemesan,
            "id_kereta":self.__id_kereta,
            "no_kursi":self.__no_kursi,
            "jadwal":self.__jadwal,
            "kelas":self.__kelas
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_id_pemesan(self, id_pemesan):
        url = self.__url+"?id_pemesan="+id_pemesan
        payload = {
            "id_pemesan":self.__id_pemesan,
            "id_kereta":self.__id_kereta,
            "no_kursi":self.__no_kursi,
            "jadwal":self.__jadwal,
            "kelas":self.__kelas
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_id_pemesan(self,id_pemesan):
        url = self.__url+"?id_pemesan="+id_pemesan
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text
